Makefile update

danmois opened the pull request:
Implementation of first makefile features regarding the use of a specific SDK.

amiclaus reviewed (commented):
It looks like these changes broke the `Build Projects` pipeline.
If you add `staging/` prefix to your branch: https://github.com/analogdevicesinc/no-OS/blob/f2c0ffea809196425300d2cdc39174959c54493f/build_projects.yml#L4
 the builds will be triggered on your custom branch, prior to doing a PR so you can debug the issues easier and double check that everything works fine :).

amiclaus commented on tools/scripts/build_projects.py:
```diff
@@ -192,11 +192,11 @@ def build(self):
+		err = run_cmd(cmd + ' update')
 		if err != 0:
 			return err
 		err = run_cmd(cmd + ' -j%d re' % (multiprocessing.cpu_count() - 1))
```
`make re` was removed with d71c363baa61b574042e58f3a037ef32d350c02e

buha replied:
yes, true, we need to make sure all the changes in this PR don't break our CI builds

danmois replied:
Done

buha commented on tools/scripts/generic.mk:
```diff
@@ -379,18 +380,11 @@ clean_all: clean_libs
 	@$(call print,[Delete] $(BUILD_DIR))
 	$(MUTE) $(call remove_dir,$(BUILD_DIR)) $(HIDE)
 
-PHONY += ca
```
I would rephrase the commit title to "...: remove obsolete/unneeded rules" or something like that

danmois replied:
Done

buha commented on tools/scripts/generic.mk:
```diff
@@ -366,7 +367,10 @@ standalone:
 
 # Build project using SDK
-project_build: $(PLATFORM)_project_build
+sdkbuild: $(PLATFORM)_project_build
```
Let's also rename $(PLATFORM)_project_build ---> $(PLATFORM)_sdkbuild so that it's easier to follow the script. This is internal-only, users wouldn't care, but it makes these scripts more maintainable... same for $(PLATFORM)_project_clean ---> $(PLATFORM)_sdkclean

danmois replied:
Done

buha reviewed (commented):
I also think we should do the changes stm32.mk and aducm3029.mk platforms in this same PR, define the rules $(PLATFORM)_sdkbuild, $(PLATFORM)_sdkclean, $(PLATFORM)_sdkopen in all of them.
If we only do it for xilinx it will fail when you `make sdkbuild` on any other platform.

danmois commented:
V2

- Removed also the ra rule

danmois commented:
V3

- renamed $(PLATFORM)_project_build to $(PLATFORM)_sdkbuild
- renamed $(PLATFORM)_project_clean to $(PLATFORM)_sdkclean

adrimbarean commented on tools/scripts/generic.mk:
```diff
@@ -375,8 +375,8 @@ clean:
 # Remove the whole build directory
-PHONY += clean_all
-clean_all: clean_libs
+PHONY += reset
```
Is "reset" the correct name here? To me "reset" implies that after the clean operation is done the build starts again. Would "purge" be a better choice?

danmois replied:
I would say that "purge" is a term that is usual for the removal of sensitive data, while "reset" implies the return to the initial state. However, I would be happy to organize a poll on this matter, and choose the proper naming.

adrimbarean replied:
No need. Whatever we choose it's gonna be great.

adrimbarean commented on tools/scripts/platform/xilinx/build_project.tcl:
```diff
@@ -5,6 +5,6 @@ if { $argc != 1 } {
 sdk setws [lindex $argv 0]
 
-projects -build -name app -type app
+app build -name app
```
I would either make a visible note about this change in the commit description or place it in a separate commit. It probably is best if placed in a separate commit because, if I'm not mistaken, this it an update to the make process following an update to the Xilinx tools. I'd say it should special notice.

danmois replied:
Yes, this is true. It is the current recommended (by the Xilinx tools) way of dealing with projects.

adrimbarean reviewed (commented):
A couple of questions.

danmois commented:
V4

- addition of the sdkclean rule for the STM32 platform
- correction of echo message of remove file function in genetic.mk
- the creation of SDK-related rules for the ADuCM3029 platform

buha commented on tools/scripts/build_projects.py:
```diff
@@ -192,14 +192,17 @@ def build(self):
 		
 		if new_hdf:
-			err = run_cmd(cmd + ' clean_all')
+			err = run_cmd(cmd + ' reset')
```
the changes in build_projects should probably go in the 2nd commit, that's the one that deals with reset and update

danmois replied:
Done

buha commented on tools/scripts/stm32.mk:
```diff
@@ -163,8 +163,8 @@ post_build: $(HEX)
 
-PHONY += $(PLATFORM)_project_build
-$(PLATFORM)_project_build: $(PROJECT_TARGET) $(LIB_TARGETS)
+PHONY += $(PLATFORM)_sdkbuild
```
commit title suggests this is for Xilinx only but you have a stm32 related change here.

buha replied:
Actually I think you should rather do each rule like you did it in the following commit:
`tools: scripts: *.mk: sdkopen rule `
Put the top-level generic.mk rule altogether with all platform specific rules that it uses in the same commit. That way, when somebody checks-out this commit, the rule will work for any platform.

danmois commented:
V5

- moved the modifications performed in build_projects.py to the second commit

danmois commented:
V6

- reorganized modifications in commits dealing with specific rules on all applicable targets (as recommended in https://github.com/analogdevicesinc/no-OS/pull/1222#discussion_r795409130)

adrimbarean reviewed (approved):
Looks good to me.

buha reviewed (approved):
Looks good, but we need to do some testing

buha commented on tools/scripts/aducm.mk:
```diff
@@ -283,4 +283,12 @@ clean_project:
 
+PHONY += $(PLATFORM)_sdkclean
+$(PLATFORM)_sdkclean:
+	$(call print,[Delete] SDK created artefacts from $(BUILD_DIR))
```
Could you please delete the "created" word ?
It generates this output and I find it very confusing whether something was created or deleted, even though by careful reading one may understand the proper meaning of it.
```
[iio_demo]$ make sdkclean
[09:52:42] [Delete] SDK created artefacts from /home/dari/Work/no-OS/projects/iio_demo/build/app/Debug
[09:52:42] [Delete] SDK created artefacts from /home/dari/Work/no-OS/projects/iio_demo/build/app/Release
```

buha commented on tools/scripts/stm32.mk:
```diff
@@ -168,6 +170,13 @@ $(PLATFORM)_sdkbuild:
 
+PHONY += $(PLATFORM)_sdkclean
+$(PLATFORM)_sdkclean:
+	$(call print,[Delete] SDK created artefacts from $(DEBUG_DIR))
```
also here, "created"
and everywhere else

buha commented:
So I've tested this on stm32 and works as expected

danmois commented:
V7

- Fixed the message displayed by the sdkclean rule (removal of the "created" work)

buha commented:
remove_dir_action on linux is missing

buha commented:
let's also implement `make build`

danmois commented:
V8

- added the build rule
- fixed a bug regarding the sdkbuild for STM32 on Linux (the addition of the remove_dir_action function in generic.mk - required for consistency with the Win implementation)

APorumb commented:
I've tested building xilinx project on windows machine and works as expected.
